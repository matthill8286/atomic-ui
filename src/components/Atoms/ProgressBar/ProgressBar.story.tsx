import { boolean, number } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { ProgressBar } from './ProgressBar'

storiesOf('Design System/Atoms/ProgressBar', module).add('Default', () => {
  const value = number('Value', 50)
  const small = boolean('Small Version', false)

  return <ProgressBar value={value} small={small} />
})
