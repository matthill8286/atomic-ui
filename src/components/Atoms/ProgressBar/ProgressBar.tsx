import * as React from 'react'
import { styled } from '@/styles/styled'

export interface ProgressBarProps {
  value: number
  max?: number
  small?: boolean
}

export interface StyledProgressProps {
  isSmall: boolean
}

const StyledProgressBar = styled.progress<StyledProgressProps>`
  display: inline-block;
  appearance: none;
  position: relative;
  height: ${({ isSmall }) => (isSmall ? '4px' : '15px')};
  border: none;
  background: ${({ theme }) => theme.color.grey2};
  background-size: auto;
  width: 100%;
  overflow: hidden;
  color: ${({ theme }) => theme.color.primary};

  &::-webkit-progress-value {
    background: ${({ theme }) => theme.color.primary};
  }

  &::-webkit-progress-bar {
    background: ${({ theme }) => theme.color.grey2};
  }

  &::-moz-progress-bar {
    background: ${({ theme }) => theme.color.primary};
  }

  &::-ms-fill {
    background: ${({ theme }) => theme.color.primary};
  }
`

export const ProgressBar: React.FC<ProgressBarProps> = ({
  value,
  max = 100,
  small = false,
  ...otherProps
}): JSX.Element => <StyledProgressBar max={max} value={value} isSmall={small} {...otherProps} />
