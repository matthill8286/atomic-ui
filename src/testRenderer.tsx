import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { saiyanTheme } from '@/styles/sc-vars-saiyan'
import { Theme } from '@/types'

export function renderWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>,
  theme = saiyanTheme
) {
  return renderer.create(<ThemeProvider theme={theme}>{tree}</ThemeProvider>).toJSON()
}

export function renderWithThemeAndRouter(tree: React.ReactElement, theme: Theme = saiyanTheme) {
  return renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>{tree}</Router>
      </ThemeProvider>
    )
    .toJSON()
}

export function renderWithRouter(tree: React.ReactElement) {
  return renderer.create(<Router>{tree}</Router>).toJSON()
}

export function mountWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(<ThemeProvider theme={saiyanTheme}>{tree}</ThemeProvider>)
}

export function mountWithThemeAndRouter<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(
    <ThemeProvider theme={saiyanTheme}>
      <Router>{tree}</Router>
    </ThemeProvider>
  )
}
