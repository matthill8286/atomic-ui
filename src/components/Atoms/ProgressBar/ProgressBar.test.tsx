import React from 'react'
import { renderWithTheme } from '@/testRenderer'
import { ProgressBar } from './ProgressBar'

describe('ProgressBar', () => {
  it('renders basic', () => {
    const tree = renderWithTheme(<ProgressBar value={12} />)
    expect(tree).toMatchSnapshot()
  })
})
