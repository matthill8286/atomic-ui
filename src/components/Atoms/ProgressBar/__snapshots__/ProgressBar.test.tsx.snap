// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders basic 1`] = `
.c0 {
  display: inline-block;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  height: 15px;
  border: none;
  background: #E5E5E5;
  background-size: auto;
  width: 100%;
  overflow: hidden;
  color: #00C8DC;
}

.c0::-webkit-progress-value {
  background: #00C8DC;
}

.c0::-webkit-progress-bar {
  background: #E5E5E5;
}

.c0::-moz-progress-bar {
  background: #00C8DC;
}

.c0::-ms-fill {
  background: #00C8DC;
}

<progress
  className="c0"
  max={100}
  value={12}
/>
`;
